src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Extensions;
+
+static class NamespaceSymbolExtensions
+{
+	public static IEnumerable<INamedTypeSymbol> GetNamedTypeSymbols(this INamespaceSymbol namespaceSymbol)
+	{
+		var stack = new Stack<INamespaceSymbol>();
+		stack.Push(namespaceSymbol);
+
+		while (stack.Count > 0)
+		{
+			var namespaceSymbolFromStack = stack.Pop();
+
+			foreach (var member in namespaceSymbolFromStack.GetMembers())
+			{
+				if (member is INamespaceSymbol memberAsNamespace)
+				{
+					stack.Push(memberAsNamespace);
+				}
+				else if (member is INamedTypeSymbol memberAsNamedTypeSymbol)
+				{
+					yield return memberAsNamedTypeSymbol;
```

Comment:
Is it intended to not return possible nested types?

Reply:
We aren't using this method anymore, I'll remove it in a future commit